projects/plugins/boost/app/modules/optimizations/lcp/class-lcp-optimize-bg-image.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Optimizations\Lcp;
+
+use Automattic\Jetpack\Image_CDN\Image_CDN_Core;
+
+class LCP_Optimize_Bg_Image {
+	/**
+	 * The LCP data for optimizing the current page.
+	 *
+	 * @var array
+	 */
+	private $lcp_data;
+
+	public static function init() {
+		if ( LCP_Optimization_Util::should_skip_optimization() ) {
+			return;
+		}
+
+		$storage  = new LCP_Storage();
+		$lcp_data = $storage->get_current_request_lcp();
+
+		if ( empty( $lcp_data ) ) {
+			return;
+		}
+
+		$instance = new self( $lcp_data );
+
+		// Preload the background image as early as possible.
+		add_action( 'wp_head', array( $instance, 'preload_background_images' ), 1 );
+
+		// Add the background image styling as late as possible.
+		add_action( 'wp_body_open', array( $instance, 'add_bg_style_override' ), 999999 );
+	}
+
+	public function __construct( $lcp_data ) {
+		$this->lcp_data = $lcp_data;
+	}
+
+	public function preload_background_images() {
+		$selectors = array();
+
+		foreach ( $this->lcp_data as $lcp_data ) {
+			if ( in_array( $lcp_data['element'], $selectors, true ) ) {
+				// If we already printed the styling for this element, skip it.
+				continue;
+			}
+			$selectors[] = $lcp_data['element'];
+
+			$responsive_image_rules = $this->get_responsive_image_rules( $lcp_data );
+			$this->print_preload_links( $responsive_image_rules );
+		}
+	}
+
+	private function print_preload_links( $responsive_image_rules ) {
```

Comment:
Not a blocker, but maybe we'd have an opportunity to switch to using the `wp_preload_resources` hook for things like that in the future?

Reply:
I wasn't aware of the hook. Let's handle it in a different PR